bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/handlers/ParadoxAlarmHandlerFactory.java

```diff
@@ -43,7 +50,9 @@
     private final Logger logger = LoggerFactory.getLogger(ParadoxAlarmHandlerFactory.class);
 
     private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = new HashSet<>(Arrays
-            .asList(COMMUNICATOR_THING_TYPE_UID, PANEL_THING_TYPE_UID, PARTITION_THING_TYPE_UID, ZONE_THING_TYPE_UID));
+        .asList(COMMUNICATOR_THING_TYPE_UID, PANEL_THING_TYPE_UID, PARTITION_THING_TYPE_UID, ZONE_THING_TYPE_UID));
+
+    private Map<ThingUID, @Nullable ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();

```

Comment:
Can be a final property.

Reply:
Yes it can and it was but I agreed with lolodomo to make it non-final. Maybe it's easier understandable that it can be changed during runtime if it's not final. What do you think?

Reply:
I have the feeling it is different: The final keyword makes the intent of the code clear to any developer. Making a variable final relives the programmer to scan the code to see if the variable has changed anywhere after its initialization. It is also much safer than non-final variables in terms of maintaining states in a multithreaded environment (see https://dzone.com/articles/all-about-finals for further information).

Keep in mind that we have distinguish between final and immutability.

Reply:
Hi,
Thanks for the link. I know very well the difference between final and immutable. 
Before I change it again I would prefer that the other reviewer @lolodomo shares his thoughts because based on his review I made the final regs collection non-final. 

Best regards,
Konstantin

Reply:
Sorry for the confusion I introduced. If I correctly understand, the map is not updated but its elements are and so the variable can be final? 
Of course you can re-add final keyword if this is better.